src/com/xilinx/rapidwright/edif/EDIFTools.java

```diff
@@ -1485,4 +1488,62 @@ Iterable<Entry<T,U>> sortIfStable(Map<T,U> collection, boolean stable) {
         }
         return collection.entrySet().stream().sorted(Entry.comparingByKey())::iterator;
     }
+
+    /**
+     * Locks the netlist by applying the DONT_TOUCH property to instances and nets
+     * so that Vivado won't make changes to it during opt_design, place_design,
+     * phys_opt_design, or route_design. Note: RapidWright doesn't respect the
+     * DONT_TOUCH attribute.
+     * 
+     * @param netlist The netlist to lock
+     */
+    public static void lockNetlist(EDIFNetlist netlist) {
+        EDIFPropertyValue value = new EDIFPropertyValue("true", EDIFValueType.BOOLEAN);
+        lockNetlist(netlist, i -> i.addProperty(DONT_TOUCH, value));
+    }
+
+    /**
+     * Unlocks the netlist by removing the DONT_TOUCH property on instances and nets
+     * so that Vivado won't make changes to it during opt_design, place_design,
+     * phys_opt_design, or route_design.
+     * 
+     * @param netlist The netlist to unlock
+     */
+    public static void unlockNetlist(EDIFNetlist netlist) {
+        lockNetlist(netlist, i -> i.removeProperty(DONT_TOUCH));
+    }
+
+    /**
+     * Traversal helper method to {@link #lockNetlist(EDIFNetlist)} and
+     * {@link #unlockNetlist(EDIFNetlist)}.
+     * 
+     * @param netlist    The netlist on which to apply the action.
+     * @param lockAction The method of action (lock or unlock) to apply to
+     *                   {@link EDIFPropertyObject} objects.
+     */
+    private static void lockNetlist(EDIFNetlist netlist, Function<EDIFPropertyObject, EDIFPropertyValue> lockAction) {
```

Comment:
Looks like this lambda doesn't need to return anything?

```suggestion
    private static void lockNetlist(EDIFNetlist netlist, Consumer<EDIFPropertyValue> lockAction) {
```